docs/4.4/api-reference.md

```diff
@@ -113,50 +113,200 @@ func main() {
+
+You may want to use the API to manage roles if:
+
+- You want to write a program that can always ensure certain roles exist on your system and do not want to orchestrate tctl to do this.
+- You want to dynamically create short lived roles.
+- You want to dynamically create roles with fields filled that Teleport currently does not support.
+
+### The Role Object
+
+Before we break down the Role object piece by piece, You can look at the `Create Role` section below to see an example of a role.
+
+A Teleport `role` is primarily defined by its `Spec`, which contains its `Allow` rules, `Deny` rules, and OpenSSH `Options`.
 
 ```go
-role, err := client.GetRole("auditor")
+// RoleV3 represents role resource specification
+type RoleV3 struct {
+  // Spec is the role specification
+  Spec RoleSpecV3 struct {
+    // Options is for OpenSSH options like agent forwarding.
+    Options RoleOptions
+    // Allow is the set of conditions evaluated to grant access.
+    Allow RoleConditions
+    // Deny is the set of conditions evaluated to deny access. Deny takes priority over allow.
+    Deny RoleConditions
+  }
+  // MetaData and other descriptors
+  ...
+}
+```
+
+**Role Options**
+
+The `RoleOptions` struct defines what OpenSSH actions a user is allowed to use. Below are a few of the most common fields, but check [here](enterprise/ssh-rbac.md#role-options) for a full list of supported options.
+
+```go
+// RoleOptions is a set of role options
+type RoleOptions struct {
+  // ForwardAgent is SSH agent forwarding. default true.
+  ForwardAgent Bool
+  // MaxSessionTTL defines how long a SSH session can last for.
+  MaxSessionTTL Duration
+  // PortForwarding defines if the certificate will have "permit-port-forwarding" in the
+  // certificate. PortForwarding is "yes" if not set.
+  PortForwarding *BoolOption // struct { Value bool } - Nullable boolean
+}
+```
+
+**Role Conditions**
+
+The `RoleConditions` struct allows for precise permission combinations. You can define what *nix logins a role can use, what resources it can read/write to, what nodes it has access to (via NodeLabels), and more.
+
+Most roles will only need to define a few of these fields.
+
+```go
+// RoleConditions is a set of conditions that must all match to be allowed or denied access.
+type RoleConditions struct {
+  // Logins is a list of *nix system logins,  e.g. "root".
+  Logins []string
+  // Namespaces is a list of namespaces (used to partition a cluster).
+  Namespaces []string
+  // NodeLabels is a map of node labels (used to dynamically grant access to nodes).
+  NodeLabels Labels // essentially map[string][]string
+  // Rules is a list of rules and their access levels. Rules represents allow or deny rule
+  // that is executed to check if user or service have access to resource
+  Rules []Rule struct {
+    // Resources is a list of resources - e.g. "roles", "users"
```

Comment:
People have been asking for an exact list of all the resources we support for quite a while - it'd be great to extract that from the code and get this into these docs somewhere.